ratis-server/src/main/java/org/apache/ratis/server/impl/LeaderElection.java

```diff
@@ -138,6 +138,7 @@ static class Executor {
 
     Executor(Object name, int size) {
       Preconditions.assertTrue(size > 0);
+      // TODO(jiacheng): intentionally ignoring the server ref here? Why?
       executor = Executors.newFixedThreadPool(size, r -> new Daemon(r, name + "-" + count.incrementAndGet()));
```

Comment:
Intentionally not passing the `RaftServerImpl` reference to the Daemon thread so a dead thread doesn't set the RaftServerImpl state to EXCEPTION. My intuition is that a failed prevote doesn't matter that much because it will be retried. Could someone confirm?